lib/Commands/NinjaBuildCommand.cpp

```diff
@@ -1204,33 +1214,47 @@ buildCommand(BuildContext& context, ninja::Command* command) {
         struct DepsActions : public core::MakefileDepsParser::ParseActions {
           BuildContext& context;
           NinjaCommandTask* task;
-          const std::string& path;
+          const StringRef workingDirectory;
+          const StringRef path;
           unsigned numErrors{0};
 
           DepsActions(BuildContext& context, NinjaCommandTask* task,
-                      const std::string& path)
-            : context(context), task(task), path(path) {}
+                      const StringRef workingDirectory,
+                      const StringRef path)
+            : context(context), task(task), workingDirectory(workingDirectory), path(path) { }
 
           virtual void error(const char* message, uint64_t position) override {
             context.emitError(
                 "error reading dependency file: %s (%s) at offset %u",
-                path.c_str(), message, unsigned(position));
+                path.str().c_str(), message, unsigned(position));
             ++numErrors;
           }
 
           virtual void actOnRuleDependency(const char* dependency,
                                            uint64_t length,
-                                           const StringRef
-                                             unescapedWord) override {
-            context.engine.taskDiscoveredDependency(task, unescapedWord);
+                                           const StringRef unescapedWord) override {
+
+            StringRef path;
+            SmallString<256> absPathTmp;
```

Comment:
Should this be the equivalent of  `PATH_MAX`?

Reply:
I don't think so:

1. PATH_MAX is 1k on macOS, which is **not enough** to fully contain the maximum path name on APFS and other file systems.
2. This thing gets cleared in the beginning, which is O(N) if we're not extremely lucky.